content/documentation/pages/6-recipes/2-file-ingest/1-sftp-to-jdbc.md

```diff
@@ -0,0 +1,936 @@
+║-spring.datasource.driverClassName(STRING) │org.mariadb.jdbc.Driver            ║
+║localFilePath(STRING)                      │/staging/shared-files/name-list.csv║
+║-spring.datasource.url(STRING)             │******                             ║
+╚═══════════════════════════════════════════╧═══════════════════════════════════╝
+```
+
+#### Verify data
+
+When the the batch job runs, it processes the file in the local directory `/staging/shared-files` and transforms each item to uppercase names and inserts it into the database.
+
+Open a shell in the `mysql` container to query the `people` table.:
+
+```bash
+$kubectl get pods
+...
+my-release-mysql-56f988dd6c-qlm8q                1/1     Running
+...
+```
+
+<!-- Rolling my own to disable erroneous formating -->
+<div class="gatsby-highlight" data-language="bash">
+<pre class="language-bash"><code>$ kubectl exec -it my-release-mysql-56f988dd6c-qlm8q -- /bin/bash
+# mysql -u root -p$MYSQL_ROOT_PASSWORD
+mysql&gt; select * from dataflow.people;
+</code></pre></div>
+
+```
++-----------+------------+-----------+
+| person_id | first_name | last_name |
++-----------+------------+-----------+
+|         1 | AARON      | AABERG    |
+|         2 | AARON      | AABY      |
+|         3 | ABBEY      | AADLAND   |
+|         4 | ABBIE      | AAGAARD   |
+|         5 | ABBY       | AAKRE     |
+|         6 | ABDUL      | AALAND    |
+|         7 | ABE        | AALBERS   |
+|         8 | ABEL       | AALDERINK |
+|         9 | ABIGAIL    | AALUND    |
+|        10 | ABRAHAM    | AAMODT    |
+|      ...                           |
++-----------+------------+-----------+
+```
+
+## Limiting concurrent task executions
+
+This recipe processes a single file with 5000+ items. What if we drop 100 files to the remote directory?
+The `sftp` source will process them immediately, generating 100 task launch requests. The Dataflow Server launches tasks asynchronously so this could potentially overwhelm the resources of the runtime platform.
+For example, when running the Data Flow server on your local machine, each launched task creates a new JVM. In Cloud Foundry, each task creates a new container instance, and in Kubernetes a pod.
+
+Fortunately, Spring Cloud Data Flow provides configuration settings to [limit the number of concurrently running tasks](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#spring-cloud-dataflow-task-limit-concurrent-executions)
+
+We can use this sample to see how this works.
+
+### Lower the maximum concurrent task executions
+
+The sample project includes 20 files in the `data/spilt` directory. To observe the limit in action we can set the maximum concurrent tasks to 3.
+
+For running tasks on a local server, restart the server, adding a command line argument `spring.cloud.dataflow.task.platform.local.accounts[default].maximum-concurrent-tasks=3`.
+
+If running on Cloud Foundry:
```

Comment:
It would be good to indicate that a "restart" of SCDF is required after setting this env-var.

Reply:
The cf CLI already provides the `restage` hint already.  If they get this far, no need to explain that here.